1. Run the NanoFLORA (https://github.com/jadedavis5/NanoFLORA) pipeline for read pre-processing and alignment
2. Take the sample to genome BAM files from the output/processing/minimap2/ output directory
3. Input these into the bambu.R script for processing
4. Take the uncleaned output files and put them through the NanoFLORA clean-stats mode
